pkg/util/pki/csr_test.go

```diff
@@ -553,3 +558,117 @@ func Test_buildKeyUsagesExtensionsForCertificate(t *testing.T) {
 		})
 	}
 }
+
+func TestSignCSRTemplate(t *testing.T) {
+	// We want to be testing the behavior of SignCSRTemplate in various
+	// contexts; for that, we construct a chain of three certificates: a root
+	// CA, an intermediate CA, and a leaf certificate.
```

Comment:
would it be helpful to test a chain of 4 certificates? e.g. with the following structure:

```text
root -> intermediate 1 -> intermediate 2 -> leaf
```

this could help to test that the chain is being handled correctly; we can assert that we end up with:

```text
ca.crt: root
tls.crt: leaf (followed by) intermediate 2 (followed by) intermediate 1
```

Reply:
Sure! I'll fix.

Reply:
Done